### PR TITLE
OPT - fix docstring and improve tests slighly

### PR DESCRIPTION
**Pull request by patrickvonplaten**
# What does this PR do?

Some final fixes for OPT that we forgot yesterday

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [ ] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#start-contributing-pull-requests),
      Pull Request section?
- [ ] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

### CODE REVIEW

File: utils/documentation_tests.txt [hunk 1]
```diff
@@ -35,6 +35,7 @@ src/transformers/models/marian/modeling_marian.py
 src/transformers/models/mbart/modeling_mbart.py
 src/transformers/models/mobilebert/modeling_mobilebert.py
 src/transformers/models/mobilebert/modeling_tf_mobilebert.py
+src/transformers/models/opt/modeling_opt.py
```
Comment (patrickvonplaten) [inline utils/documentation_tests.txt:38]: @ArthurZucker we should always try to test our docstrings to be sure they work when people look into the docs :-)

File: tests/models/opt/test_modeling_opt.py [hunk 2]
```diff
@@ -330,33 +330,61 @@ def test_logits(self):
         assert torch.allclose(logits, logits_meta, atol=1e-4)
 
 
-@require_tokenizers
```
Comment (patrickvonplaten) [inline tests/models/opt/test_modeling_opt.py:333]: no need for the tokenizers library here as we only have the slow one for now

File: tests/models/opt/test_modeling_opt.py [hunk 3]
```diff
@@ -330,33 +330,61 @@ def test_logits(self):
...
-        NEXT_TOKENS = [3392, 764, 5, 81]
-        GEN_OUTPUT = []
 
-        tokenizer = GPT2Tokenizer.from_pretrained("patrickvonplaten/opt_gpt2_tokenizer")
-        for model in self.all_model_path:
-            model = OPTForCausalLM.from_pretrained(self.path_model)
-            model = model.eval()
-            model.config.eos_token_id = tokenizer.eos_token_id
-
-            gen = pipeline("text-generation", model=model, tokenizer=tokenizer, return_tensors=True)
```
Comment (patrickvonplaten) [inline tests/models/opt/test_modeling_opt.py:356]: better not use pipeline in modeling tests, this creates unnecessary dependencies (also for the test fetcher)

**Review by sgugger — APPROVED** [review]
Thanks a lot for all the fixes!

File: utils/documentation_tests.txt (hunk 1 above)
Comment (ArthurZucker) [reply utils/documentation_tests.txt:38]: Oh that's where you put it !

**Review by ArthurZucker — APPROVED** [review]
Let's merge!

**Pull request merged**